edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -117,14 +129,27 @@ protected internal async Task<T> Execute<T>(Func<Task<T>> func, string operation
             try
             {
                 Events.ExecutingOperation(operation, this.ManagementUri.ToString());
-                T result = await ExecuteWithRetry(func, r => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r), this.transientErrorDetectionStrategy)
+                T result = await ExecuteWithRetry(
+                    func,
+                    (r) => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r),
+                    this.transientErrorDetectionStrategy)
                     .TimeoutAfter(this.operationTimeout);
                 Events.SuccessfullyExecutedOperation(operation, this.ManagementUri.ToString());
+                this.failedSocketCount = 0;
                 return result;
             }
             catch (Exception ex)
             {
                 Events.ErrorExecutingOperation(ex, operation, this.ManagementUri.ToString());
+                if (this.IsModuleManagementSocketStale(ex))
+                {
+                    this.failedSocketCount++;
+                    if (this.failedSocketCount >= MaxRetryCount)
+                    {
+                        Environment.Exit((ex as SocketException)?.ErrorCode ?? 110);
```

Comment:
You need to log error message to make sure we see the reason why edgeHub exits.

Reply:
It is already taken care by the `Event.Error()` as the first thing exception is thown.